app/controllers/registrations_controller.rb

```diff
@@ -591,12 +606,20 @@ def payment_completion
     stored_intent.update_status_and_charges(stripe_intent, current_user) do |charge_transaction|
       ruby_money = charge_transaction.money_amount
 
-      registration.record_payment(
-        ruby_money.cents,
-        ruby_money.currency.iso_code,
-        charge_transaction,
-        current_user.id,
-      )
+      if uses_v2
+        begin
+          Microservices::Registrations.update_registration_payment(attendee_id, charge.id, ruby_money.cents, ruby_money.currency.iso_code, stripe_intent.status)
+        rescue Faraday::Error
+          return redirect_to competition_register_path(competition_id, "registration_unreachable")
```

Comment:
Same old issue with redirect: This will redirect you to /competition/register.registration_unreachable` which probably does not do what you want. Set `flash[:error]` instead.

Reply:
copy paste whoopsie